R/preprocessing_io.R

```diff
@@ -750,7 +750,7 @@ fread_maf = function(maf_file_path){
                      skip = "Hugo_Symbol",
                      quote = "",
-                     colClasses=c(Chromosome="character",PHENO="character",PUBMED="character",gnomADg_AF="character",SOMATIC="character",HGNC_ID="character"))
+                     colClasses=c(Hugo_Symbol="character",Entrez_Gene_Id="integer",Center="character",NCBI_Build="character",Chromosome="character",Start_Position="integer",End_Position="integer",Strand="character",Variant_Classification="character",Variant_Type="character",Reference_Allele="character",Tumor_Seq_Allele1="character",Tumor_Seq_Allele2="character",dbSNP_RS="character",dbSNP_Val_Status="logical",Tumor_Sample_Barcode="character",Matched_Norm_Sample_Barcode="character",Match_Norm_Seq_Allele1="character",Match_Norm_Seq_Allele2="character",Tumor_Validation_Allele1="logical",Tumor_Validation_Allele2="logical",Match_Norm_Validation_Allele1="logical",Match_Norm_Validation_Allele2="logical",Verification_Status="logical",Validation_Status="logical",Mutation_Status="logical",Sequencing_Phase="logical",Sequence_Source="logical",Validation_Method="logical",Score="logical",BAM_File="logical",Sequencer="logical",Tumor_Sample_UUID="logical",Matched_Norm_Sample_UUID="logical",HGVSc="character",HGVSp="character",HGVSp_Short="character",Transcript_ID="character",Exon_Number="character",t_depth="integer",t_ref_count="integer",t_alt_count="integer",n_depth="integer",n_ref_count="integer",n_alt_count="integer",all_effects="character",Allele="character",Gene="character",Feature="character",Feature_type="character",Consequence="character",cDNA_position="character",CDS_position="character",Protein_position="character",Amino_acids="character",Codons="character",Existing_variation="character",ALLELE_NUM="integer",DISTANCE="integer",STRAND_VEP="integer",SYMBOL="character",SYMBOL_SOURCE="character",HGNC_ID="character",BIOTYPE="character",CANONICAL="character",CCDS="character",ENSP="character",SWISSPROT="character",TREMBL="character",UNIPARC="character",RefSeq="character",SIFT="character",PolyPhen="character",EXON="character",INTRON="character",DOMAINS="character",AF="character",AFR_AF="character",AMR_AF="character",ASN_AF="logical",EAS_AF="character",EUR_AF="character",SAS_AF="character",AA_AF="character",EA_AF="character",CLIN_SIG="character",SOMATIC="character",PUBMED="character",MOTIF_NAME="logical",MOTIF_POS="logical",HIGH_INF_POS="logical",MOTIF_SCORE_CHANGE="logical",IMPACT="character",PICK="integer",VARIANT_CLASS="character",TSL="logical",HGVS_OFFSET="integer",PHENO="character",MINIMISED="logical",GENE_PHENO="integer",FILTER="character",flanking_bps="character",vcf_id="character",vcf_qual="character",gnomAD_AF="logical",gnomAD_AFR_AF="logical",gnomAD_AMR_AF="logical",gnomAD_ASJ_AF="logical",gnomAD_EAS_AF="logical",gnomAD_FIN_AF="logical",gnomAD_NFE_AF="logical",gnomAD_OTH_AF="logical",gnomAD_SAS_AF="logical",vcf_pos="integer",gnomADg_AF="character",blacklist_count="integer"))
   return(maf_dt)
```

Comment:
For readability and editability, it's much better to break up long lists of parameters onto multiple lines. In fact the R style guide says one parameter per line, and no lines should be longer than 80 characters. Some style tips here: https://style.tidyverse.org/syntax.html#syntax

I tried loading in one of the CDS mafs with read_tsv, and the col_types for several of the columns don't match what you have here. You can check it with the following command: 
```
spec(read_tsv("results/all_the_things/slms_3-1.0_vcf2maf-1.3/capture--projection/deblacklisted/maf/all_slms-3--grch37.CDS.maf"))
```

If we updated this function to use `read_tsv` instead of `fread`, you could do this: 
```
# Get a minimal specification of col_types by reading in the CDS maf file 
maf_col_types <- cols_condense(spec(read_tsv("results/all_the_things/slms_3-1.0_vcf2maf-1.3/capture--projection/deblacklisted/maf/all_slms-3--grch37.CDS.maf")))
# Save this as an object to the data directory
save(maf_col_types, file = "data/maf_col_types.rda")

# In all future instances of fread_maf or any other maf-loading function (including `get_ssm_by_samples(subset_from_merge = TRUE)`, you can use this object
maf_col_types <- load("data/maf_col_types.rda")
fread_maf <- function(maf_file_path){
  maf_dt <- readr::read_tsv(
    maf_file_path, 
    ... 
    col_types = maf_col_types
  )
)
}
```

Reply:
Any creation of an rda file like this should happen in this R script for reproducibility:
data-raw/DATASET.R

Reply:
Adam you should familiarize yourself with that script since it tracks how every rda file (shared object) in GAMBLR came to be. I would add that any file that script uses should ALSO exist in the repo (you can see examples in that script). Hence, any demo MAF snippit really should be put into the repo if it will be used by this script.

Reply:
Thanks for your comment @lkhilton. Yes, I thought about this yesterday too and I think the reason that I got some of the column types wrong was because I took a random maf and checked the column types of it and all columns that had NA most likely got interpreted as logical. It does not seem to be a problem with fread, since the function only overrides coltypes of a higher type. But nonetheless, the column types specified here should match the expected type. I will update the column types as well as improve the readability.

Reply:
Sounds good @rdmorin I'll go over this script. Thanks,

Reply:
Ryan and I debated this and I think this is an instance where generating this .rda object doesn't necessarily have to be done through the DATASET.R script, and I don't think it's worth adding a maf snippet to this repository just to be able to reproduce that object. It's still good for you to see how the DATASET.R script works though.

Reply:
I have now updated the column classes that were not a match. Mostly, columns that wrongly got classified as logical because of missing values in the template MAF I used to interrogate column classes. I have also improved the readability as suggested. In my next PR, I can work towards switching from fread to read_tsv as detailed above, if we think this is a good strategy.